transport/http/outbound.go

```diff
@@ -168,8 +183,70 @@ func (o *outbound) Call(ctx context.Context, treq *transport.Request) (*transpor
 	return nil, getErrFromResponse(response)
 }
 
-func (o *outbound) getPeerForRequest(ctx context.Context, treq *transport.Request) (*hostport.Peer, error) {
-	peer, err := o.PeerList.ChoosePeer(ctx, treq)
+type ack struct {
+	time time.Time
+}
+
+func (a ack) String() string {
+	return a.time.String()
+}
+
+// CallOneway makes a oneway request
+func (o *Outbound) CallOneway(ctx context.Context, treq *transport.Request) (transport.Ack, error) {
+	if !o.started.Load() {
+		// panic because there's no recovery from this
+		panic(errOutboundNotStarted)
+	}
+
+	peer, err := o.getPeerForRequest(ctx, treq)
+	if err != nil {
+		return nil, err
+	}
+	endRequest := peer.StartRequest()
+	defer endRequest()
+
+	req, err := o.createRequest(peer, treq)
+	if err != nil {
+		return nil, err
+	}
+
+	start := time.Now()
+	var ttl time.Duration
+
+	req.Header = applicationHeaders.ToHTTPHeaders(treq.Headers, nil)
+	ctx, req, span := o.withOpentracingSpan(ctx, req, treq, start)
+	defer span.Finish()
+	req = o.withCoreHeaders(req, treq, ttl)
```

Comment:
This will send a TTL of 0 over the wire.

Maybe withCoreHeaders should skip the header if TTL is 0.

Also, note that the literal `0` can be used where `time.Duration` is expected
because it's just an alias for `int64`.